instrumentation/pekko/pekko-http-1.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/pekkohttp/v1_0/server/tapir/RouteWrapper.java

```diff
@@ -50,9 +63,7 @@ public Unit apply(Try<RouteResult> tryResult) {
                   "*",
                   Option.apply("*"),
                   Option.apply("*"));
-
-          PekkoRouteHolder.push(path);
-          PekkoRouteHolder.endMatched();
+          routeHolder.push(beforeMatch, EMPTY, path);
```

Comment:
Hrm, I'm surprised we need `beforeMatch` here from the `RequestContext`. Did you find that tapir was missing part of the URI path?

Reply:
This is needed for the (unlikely) scenario when someone combines a pekko `pathPrefix` with a tapir route, as in [this test](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/13491/files#diff-84d36501b7a91dc8ce1e20a01a388db4ef77cdf6cfd8520e3be53444a975919bR96-R110). The `beforeMatch` is used in [the algorithm](https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/13491/files#diff-54bb48022c8c47344cf46ea3533a9c4c25a58273cfac2e40a78a46d64df8e05dR36-R46) for deciding if this match is a real match, or a composite PathMatcher that is combining the results of others.

Reply:
Ah, I see, very nice. Thanks again for the explanation 🙂.